src/components/Common/CircleStep.vue

```diff
@@ -0,0 +1,75 @@
+<template>
```

Comment:
Why don't you put the entire thing in a component but just the circle?

Reply:
Doesnt seem crucial, considering its used two times.

Reply:
In that case, this is also an overkill.

If you move the entire thing in here and make it generate based on an array of steps, the code will be shorter, cleanner and less buggy